src/backend/distributed/transaction/backend_data.c

```diff
@@ -780,6 +805,113 @@ MarkCitusInitiatedCoordinatorBackend(void)
 }
 
 
+/*
+ * AssignGlobalPID assigns a global process id for the current backend if
+ * it is not already assigned
+ */
+void
+AssignGlobalPID(void)
+{
+	if (MyBackendData->globalPID != 0)
+	{
+		return;
+	}
+
+	if (!IsCitusInitiatedRemoteBackend())
+	{
+		uint64 globalPID = GenerateGlobalPID();
+
+		SpinLockAcquire(&MyBackendData->mutex);
+		MyBackendData->globalPID = globalPID;
+		SpinLockRelease(&MyBackendData->mutex);
+
+		return;
+	}
+
+	uint64 globalPID = ExtractGlobalPID();
+
+	SpinLockAcquire(&MyBackendData->mutex);
+	MyBackendData->globalPID = globalPID;
+	SpinLockRelease(&MyBackendData->mutex);
+}
+
+
+/*
+ * GetGlobalPID returns the global process id of the current backend
+ */
+uint64
+GetGlobalPID(void)
+{
+	/* assign here incase it is not assigned already */
+	AssignGlobalPID();
+
+	uint64 globalPID = 0;
+	SpinLockAcquire(&MyBackendData->mutex);
+	globalPID = MyBackendData->globalPID;
+	SpinLockRelease(&MyBackendData->mutex);
+
+
+	return globalPID;
+}
+
+
+/*
+ * GenerateGlobalPID generates the global process id for the current backend
+ */
+static uint64
+GenerateGlobalPID(void)
+{
+	/*
+	 * We try to create a human readable global pid that consists of node id and process id.
+	 * By multiplying node id with 10^10 and adding pid we generate a number where the smallest
+	 * 10 digit represent the pid and the remaining digits are the node id
+	 *
+	 * Both node id and pid are 32 bit. We use 10^10 to fit all possible pids. Some very large
+	 * node ids might cause overflow. But even for the applications that scale around 50 nodes every
+	 * day it'd take about 100K years. So we are not worried.
+	 */
+	return (((uint64) GetLocalNodeId()) * 10000000000) + getpid();
+}
+
+
+/*
+ * ExtractGlobalPID extracts the global process id from the application name and returns it
+ * if the application name is not compatible with Citus' application names returns 0
+ */
+uint64
+ExtractGlobalPID(void)
```

Comment:
I think `ExtractGlobalPID` should get `application_name` as a parameter. Otherwise, it looks vague that we *extract* anything for the readers & other parts of the code.